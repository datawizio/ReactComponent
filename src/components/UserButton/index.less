@import "~antd/es/avatar/style/index";
@import "~antd/es/dropdown/style/index";

@import "../Button/index.less";
@import "../../styles/variables";

.ant-btn.user-name {
  height: 32px;
  padding: 0;
  display: flex;
  align-items: center;
  .user-button-fullname {
    margin-left: @margin-xs;
  }
  .anticon-down {
    padding-top: 5px;
    font-size: 12px;
  }
}

.user-dropdown {
  .ant-dropdown-menu-item-group-title {
    color: rgba(0, 0, 0, 0.65);
  }
}
