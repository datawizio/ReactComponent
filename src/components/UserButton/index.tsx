import React, { useEffect, useState } from "react";

import Button from "../Button";
import UserAvatar from "../UserAvatar";
import { DownOutlined } from "@ant-design/icons";
import { Dropdown } from "antd";

import "./index.less";
import clsx from "clsx";

export interface UserButtonProps {
  photo: string;
  fullName: string;
  menu: React.ReactElement;
  showFullName?: boolean;
  theme?: "dark" | "light";
  open?: boolean;
  handleOpenChange?: (value: boolean) => void;
}

const UserButton: React.FC<UserButtonProps> = ({
  photo,
  fullName,
  menu,
  theme,
  showFullName,
  open,
  handleOpenChange
}) => {
  const className = clsx({
    "user-name": true,
    "dw-dark": theme === "dark"
  });
  const [visible, setVisible] = useState<boolean>();
  const handleVisibleChange = (value: boolean) => {
    setVisible(value);
    handleOpenChange && handleOpenChange(value);
  }
  useEffect(() => setVisible(open), [open]);

  return (
    <Dropdown
      overlay={menu}
      className="user-dropdown"
      trigger={["click"]}
      visible={visible}
      onVisibleChange={handleVisibleChange}
    >
      <Button border={false} className={className}>
        <UserAvatar src={photo} name={fullName} />
        {showFullName ? (
          <span className="user-button-fullname">{fullName}</span>
        ) : null}
        <DownOutlined />
      </Button>
    </Dropdown>
  );
};

export default UserButton;
