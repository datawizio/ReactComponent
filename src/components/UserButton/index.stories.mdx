import { Meta, Story, Preview, Props } from "@storybook/addon-docs/blocks";
import UserButton from "./index";
import UserMenu from "../UserMenu";

<Meta title="Components/UserButton" component={UserButton} />

# UserButton

Компонент

## Basic

<Preview>
  <Story name="Basic">
    <UserButton
      fullName="Olloll Trololo"
      menu={
        <UserMenu email="email.com" fullName="Stepan Bandera" />
      }
    />
  </Story>
</Preview>

## Props

<Props of={UserButton} />
