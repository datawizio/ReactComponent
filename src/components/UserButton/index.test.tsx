import "jsdom-global/register";
import React from "react";
import { shallow } from "enzyme";

import UserButton from "./index";

const mockProps = {
  photo: "photopath",
  fullName: "FullUserName",
  menu: <></>,
  showFullName: true
};

const setUp = (props?) => shallow(<UserButton {...props} />);

describe("UserButton component", () => {
  let component;
  beforeEach(() => {
    component = setUp(mockProps);
  });

  it("UserButton rendered correctly", () => {
    expect(component).toMatchSnapshot();
  });

  it("UserButton show full name", () => {
    expect(component.html()).toContain(mockProps.fullName);
  });

  it("UserButton hide full name", () => {
    const wrapper = setUp({ ...mockProps, showFullName: false });
    expect(wrapper.html()).not.toContain(mockProps.fullName);
  });

  it("UserButton without photo", () => {
    const wrapper = setUp({ ...mockProps, photo: null });
    expect(wrapper.find("img").length).toBeFalsy();
  });
});
