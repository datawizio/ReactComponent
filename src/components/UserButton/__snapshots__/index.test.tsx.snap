// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserButton component UserButton rendered correctly 1`] = `
<Dropdown
  className="user-dropdown"
  mouseEnterDelay={0.15}
  mouseLeaveDelay={0.1}
  onVisibleChange={[Function]}
  overlay={<React.Fragment />}
  trigger={
    Array [
      "click",
    ]
  }
>
  <Button
    border={false}
    className="user-name"
  >
    <UserAvatar
      name="FullUserName"
      src="photopath"
    />
    <span
      className="user-button-fullname"
    >
      FullUserName
    </span>
    <ForwardRef(DownOutlined) />
  </Button>
</Dropdown>
`;
